sw/device/silicon_creator/lib/sigverify/sphincsplus/test/sphincsplus_set_testvectors.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+
+import hjson
+from mako.template import Template
+
+'''
+Read in an HJSON test vector file, convert the test vector to C constants, and
+generate a header file with these test vectors.
+'''
+
+
+def hex_to_hexbytes(x):
+    '''Convert a hex string to a list of bytes as hex strings.'''
+    if x.startswith('0x'):
+        x = x[2:]
+
+    # Double-check that length is even
+    if len(x) % 2 != 0:
+        raise ValueError(f'Cannot convert odd-length hex string (length {len(x)}) to bytes: {x}')
+
+    out = []
+    for i in range(0, len(x), 2):
+        out.append('0x' + x[i:i + 2])
+    return out
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--hjsonfile', '-j',
+                        metavar='FILE',
+                        required=True,
+                        type=argparse.FileType('r'),
+                        help='Read test vectors from this HJSON file.')
+    parser.add_argument('--template', '-t',
+                        metavar='FILE',
+                        required=True,
+                        type=argparse.FileType('r'),
+                        help='Read header template from this file.')
+    parser.add_argument('--headerfile', '-o',
+                        metavar='FILE',
+                        required=True,
+                        type=argparse.FileType('w'),
+                        help='Write output to this file.')
+
+    args = parser.parse_args()
+
+    # Read test vectors and stringify them
+    testvecs = hjson.load(args.hjsonfile)
```

Comment:
I think this may work, and may be safer as it handles exceptions (no need to explicitly call `close`).  Same for the file operations below (`args.headerfile` and `args.template`):

```python
with args.hjsonfile as f:
  testvecs = hson.load(f)
```